Source/Tests/IO/FileIdentifier.cpp

```diff
@@ -0,0 +1,52 @@
+//
+// Copyright (c) 2022-2022 the rbfx project.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR rhs
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR rhsWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR rhs DEALINGS IN
+// THE SOFTWARE.
+//
+
+#include "../CommonUtils.h"
+
+#include <Urho3D/IO/FileIdentifier.h>
+
+TEST_CASE("FileLocator")
+{
+    FileIdentifier signleForwardSlash("file:/FileName");
+    CHECK(signleForwardSlash.scheme_ == "file");
+    CHECK(signleForwardSlash.fileName_ == "FileName");
+
+    FileIdentifier twoForwardSlash("file://FileName");
+    CHECK(twoForwardSlash.scheme_ == "file");
+    CHECK(twoForwardSlash.fileName_ == "FileName");
+
+    FileIdentifier threeForwardSlash("file:///FileName");
+    CHECK(threeForwardSlash.scheme_ == "file");
+    CHECK(threeForwardSlash.fileName_ == "FileName");
```

Comment:
While correct, we definitely do not want to allow lookup outside of mounted folders. So in this case `/FileName` should mean a file at the top level of virtual filesystem. Also, `FileName` is same as `./FileName` in general case, but since we do not have a concept of "current directory" in vfs its not really important imho. So, we need to do something with `FileName` and only logical thing is to assume its equal to `/FileName`, which is top of virtual filesystem. Still, i would call it a minor bug in `FileIdentifier`.

Reply:
Why not? Absolute file paths exist and may be useful.

Reply:
Because this is a virtual filesystem and URIs describe items within that virtual filesystem. It makes no sense to have URIs that escape VFS unexpectedly. If one so desires `C:\` or `/` can be mounted and then you can access anything anywhere

Reply:
But "file://" *is* the mount point for the entire physical file system, no?

Reply:
Its anything that we decide when we mount folders. This question is good though, if we use `file` to mount `/`, then leading slash becomes important, because application does have a concept of current working dir. So can indeed be important sometimes

Reply:
Wat. New thing i learned today :eyes:

Reply:
These are not absolute file paths. These are related to the root of the VFS, i.e. the same as Resource.
"But "file://" is the mount point for the entire physical file system, no?" No. It is invalid file URI. file:///filename would point to the root of virtual file system.

Please reopen this comment if you disagree.